src/test/java/com/puppycrawl/tools/checkstyle/AbstractModuleTestSupport.java

```diff
@@ -372,6 +377,49 @@ private static void verifyConfig(Configuration testConfig, Configuration parsedC
         }
     }
 
+    /**
+     * Performs verification of violation lines.
+     *
+     * @param config parsed config.
+     * @param file file path.
+     * @param inputConfiguration InputConfiguration object.
+     * @throws Exception if exception occurs during verification process.
+     */
+    private void verifyViolations(Configuration config,
+                                  String file,
+                                  InputConfiguration inputConfiguration) throws Exception {
+        final Map<String, List<String>> actualViolations =
+                getActualViolationsForFile(config, file);
+        final List<Integer> expectedViolationLines =
+                actualViolations.getOrDefault(file, new ArrayList<>())
+                        .stream()
+                        .map(message -> message.substring(0, message.indexOf(':')))
+                        .map(Integer::parseInt)
+                        .distinct()
+                        .collect(Collectors.toList());
+        assertWithMessage("Violation lines differ.")
+                .that(inputConfiguration.getViolations())
+                .isEqualTo(expectedViolationLines);
+    }
+
+    /**
+     * Tests the file with the check config.
+     *
+     * @param config check configuration.
+     * @param file input file path.
+     * @return map of actual violations.
+     * @throws Exception if exception occurs during verification process.
+     */
+    private Map<String, List<String>> getActualViolationsForFile(Configuration config,
+                                                                 String file) throws Exception {
+        final List<File> files = Collections.singletonList(new File(file));
+        final Checker checker = createChecker(config);
+        final Map<String, List<String>> actualViolations =
+                getActualViolations(checker.process(files));
+        checker.destroy();
+        return actualViolations;
```

Comment:
as its for single file we can directly return `List<String>`

Reply:
`return actualViolations.getOrDefault(file, new ArrayList<>());`

Reply:
done